examples/experimental/csv-parse.js

```diff
@@ -0,0 +1,24 @@
+import { open } from 'k6/experimental/fs'
+import csv from 'k6/experimental/csv'
+import { scenario } from 'k6/execution'
+
+export const options = {
+	iterations: 10,
+}
+
+// Open the csv file, and parse it ahead of time.
+let file;
+let csvRecords;
+(async function () {
+	file = await open('data.csv');
+
+	// The `csv.parse` function consumes the entire file at once, and returns
+	// the parsed records as a SharedArray object.
+	csvRecords = await csv.parse(file, {delimiter: ','})
```

Comment:
My initial reaction to your comment is positive, but to be honest I don't have a strong opinion. I see pros & cons for both approaches, either hiding the implementation details or being specific to leave the more generic methods of the API for the future (stabilization).

Reply:
I'm unsure I fully understood the initial point about shared arrays and their constructor not being async. Can you clarify what the underlying concern is, especially from the user's perspective? 🙇‍♂️

Under those premice, I have a somewhat firm opinion that we should keep the `CSV.parse` name as it does not emphasize the return type and leaves us free to change it in the future if we wish. Also, from a user perspective, `csv.parse` says what it does without any further context and ensures sensible defaults., whereas `csv.sharedArray` or alternative implies I have to have or acquire the context of what that is and why I would want to use it.

Reply:
To be honest, I don't fully see the problems mentioned above, because:

> The main point here is that this is not as users used SharedArray till now. We never had a function returning a SharedArray directly. Instead, we required them to use an array and generate it from the constructor.

I only see it as a different (new) way to initialize a `SharedArray`, but nothing changes with respect on how it is used later in the code, I guess. And, although it's true that would be better to have a way to use the `csv` module with the "standard" way of constructing a `SharedArray`, I don't see it like a problem.

> In addition, it has the downside that the user can't change/modify the array before storing it in the SharedArray. I don't know if it is a real case and if we should care about it, just raising the question.

Imho, this is likely something to work on in the future, as a second iteration of this package. Like, in general I don't see the _immutability_ of the parsed records as a problem, instead I'd just like to be able to give some further options to the users, or build some abstraction on top. Moreover, unless I'm wrong, `SharedArray`s are read-only, so it I see it as another sign that it fits the use case.

<br/>
All that said, I think it is fine enough to keep it as-is, also considering this is going to be _experimental_ for a while.